### PR TITLE
Implement take and first

### PR DESCRIPTION
**Pull request by iduartgomez**
- Adding `take` and `first`.
- Removed a couple more lints from clippy.

Don't merge yet as still have to add tests, just doing the PR for visibility. 

P.S: We may want to add some rustfmt config and force fmt on save to avoid problems with formatting on commits that add noise.

### CODE REVIEW

**Comment by iduartgomez** [comment]
Added tests so is good to merge.

File: src/rdd.rs [hunk 1]
```diff
@@ -210,7 +210,87 @@ pub trait Rdd<T: Data>: RddBase + Send + Sync + Serialize + Deserialize {
...
+            });
+
+            let res = self.get_context().run_job_on_partitions(
+                self.get_rdd(),
+                take_from_partion,
+                partitions,
+            );
+
+            let current_len = buf.len();
+            buf.extend(
```
[the remark this answers is not in the document]
Comment (iduartgomez) [reply src/rdd.rs]: The tests seem to work fine ( I added a new one though for out of bound 'take', which also works), wanted to debug it properly but as was having problems didn't get to it yesterday.

Your example seems to be working fine unless I am missing something obvious, [slightly moldified](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f7f40533521f54b30288da2855cbda55). Takes 3 iterations to take the 8 results, pushing the exact number of results for each pseudo-partition, and current_len is changing between iterations.

[the remark this answers is not in the document]
Comment (rajasekarv) [reply src/rdd.rs]: Actually, commented out part is the logic in take method. Now I changed it. You can see how it produces the wrong [result](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=357a34c9a83bb97c98534c592333854f).    
In take test, use this example, it should panic. 
let col1 = vec![1, 2, 3, 4,5,6];

[the remark this answers is not in the document]
Comment (iduartgomez) [reply src/rdd.rs]: Ahh ok I see what you mean now. You are exactly right, didn't see the commented part in the first example and wasn't very awake this morning.

Fixed it, now it works as intended. Also changed the test to check for this as suggested.

**Pull request merged**